### DISCUSSION TITLE
Using variable references can solve the problem that module hooks cannot completely change customer-defined data through displayCustomization, greatly improving system flexibility.

### DISCUSSION THREAD

**Discussion by letwang** [PrestaShop/PrestaShop#31771] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### The problem
https://github.com/PrestaShop/PrestaShop/blob/8.1.0-beta.1/classes/Product.php#L6064
```
$row['value'] = Hook::exec('displayCustomization', ['customization' => $row], (int) $row['id_module']);
```
I can't change other values of $row like name in module hook.

### The solution

```
$row['value'] = Hook::exec('displayCustomization', ['customization' => &$row], (int) $row['id_module']);
```
Pass the $row variable as a reference, you can customize any value of $row, and the module hook is highly customized.

### Alternatives

Maybe you can consider passing the second parameter of Hook::exec as a variable reference, so that the variables of the entire system can be customized and changed through the module hook.

### Additional context

```
public function hookDisplayCustomization (array &$params)
{
    $params['customization']['name'] = str_replace('a', 'b', $params['customization']['name']);
    return $params['customization']['value'];
}
```

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### Steps to reproduce

---

### PrestaShop version(s) where the bug happened

1.7.0.0 to 8.0.1

**Comment by AureRita** [comment]
Hello @letwang ,

Thank you for submitting this issue. Unfortunately, there is not enough information provided to reproduce it and work on it. **You must follow the template and provide clear and concise steps to reproduce the bug**. 
Read more about how we expect the issues to be handled [here](https://build.prestashop.com/news/how-bug-reports-are-handled/). 

Waiting for your feedback

**Comment by kpodemski** [comment]
Hi @letwang 

Let's discuss it. It's not a bug report, it's instead a discussion of an improvement. I'm transferring this issue to Discussions :)
